src/components/v5/common/CompletedAction/partials/SetUserRoles/SetUserRoles.tsx

```diff
@@ -45,11 +50,22 @@ interface Props {
 
 const transformActionRolesToColonyRoles = (
   roles: ColonyActionRoles | null | undefined,
+  historicRoles: GetColonyHistoricRoleRolesQuery['getColonyHistoricRole'],
 ): ColonyRole[] => {
   if (!roles) return [];
-  const roleKeys = Object.keys(roles);
+
+  const combinedRoles = { ...historicRoles };
+
+  for (const [key] of Object.entries(roles)) {
```

Comment:
I think we can do `[key, value]` here since we're using `Object.entries`

Reply:
@bassgeta yep but in this scenario, I only need the `key` 🔑

Reply:
@bassgeta I think my brain rotted 😂 I am in fact referencing values from `roles` ! will push a change